contracts/icp/context-config/src/mutate.rs

```diff
@@ -0,0 +1,299 @@
+use std::ops::Deref;
+
+use crate::guard::Guard;
+use crate::types::{
+    Context, ContextRequest, ContextRequestKind, ICApplication, ICCapability, ICContextId,
+    ICContextIdentity, ICPSigned, ICSignerId, Request, RequestKind,
+};
+use crate::CONTEXT_CONFIGS;
+
+#[ic_cdk::update]
+pub fn mutate(signed_request: ICPSigned<Request>) -> Result<(), String> {
+    let request = signed_request
+        .parse(|r| &r.signer_id)
+        .map_err(|e| format!("Failed to verify signature: {}", e))?;
+
+    // Check request timestamp
+    let current_time = ic_cdk::api::time();
+    if current_time.saturating_sub(request.timestamp_ms) > 1000 * 5 {
+        // 5 seconds threshold
+        return Err("request expired".to_string());
+    }
+
+    match &request.kind {
+        RequestKind::Context(ContextRequest { context_id, kind }) => match kind {
+            ContextRequestKind::Add {
+                author_id,
+                application,
+            } => {
+                add_context(
+                    &request.signer_id,
+                    context_id.clone(),
+                    author_id.clone(),
+                    application.clone(),
```

Comment:
there should be no need to clone here

```suggestion
    match request.kind {
        RequestKind::Context(ContextRequest { context_id, kind }) => match kind {
            ContextRequestKind::Add {
                author_id,
                application,
            } => {
                add_context(
                    &request.signer_id,
                    context_id,
                    author_id,
                    application,
```